### PR TITLE
Adding new metric type to track in-flight remote read queries.

### LINKED ISSUE #4661: Metric for number of remote_read queries in-flight
We have a metric that shows the number in-flight queries on the promql engine: `prometheus_engine_queries`, with the introduction of the concurrency limits #4656, a similar metric should be added for in-flight remote-read queries.

### PR DESCRIPTION
**Pull request by tariq1890**
Add new metric type to track number of remote read queries in the remote storage module of prometheus.

### CODE REVIEW

**Comment by tariq1890** [comment]
@simonpasquier Thanks! Review comments addressed.

**Comment by simonpasquier** [comment]
Thanks for the update! I'll let the last word to @gouthamve as he filed the PR originally.

File: storage/remote/read.go [hunk 1]
```diff
@@ -16,15 +16,31 @@ package remote
...
+
+func init() {
+	prometheus.MustRegister(remoteReadQueries)
+}
+
 // QueryableClient returns a storage.Queryable which queries the given
 // Client to select series sets.
 func QueryableClient(c *Client) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
+		remoteReadQueries.WithLabelValues(c.Name())
```
Comment (gouthamve) [inline storage/remote/read.go]: We can move this up, i.e,

```go
// QueryableClient returns a storage.Queryable which queries the given
// Client to select series sets.
func QueryableClient(c *Client) storage.Queryable {
	remoteReadQueries.WithLabelValues(c.Name())

	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
		return &querier{
			ctx:    ctx,
			mint:   mint,
			maxt:   maxt,
			client: c,
		}, nil
	})
}
```

**Comment by gouthamve** [comment]
Hmm, this looks good to me and should be included, but the original issue is for the read endpoint exposed by prometheus. See: https://github.com/prometheus/prometheus/blob/master/web/api/v1/api.go#L763:17

Feel free to open an new PR for it, or updating this PR. Sorry for not being clear in the issue.

File: storage/remote/read.go (hunk 1 above)
Comment (gouthamve) [reply storage/remote/read.go]: The reason for it is until we use the `Queryable` we won't be initialising the metric.

Comment (tariq1890) [reply storage/remote/read.go]: oh yes! Thanks for pointing that out.

**Comment by tariq1890** [comment]
@gouthamve Thanks for the review. I have made the review comments and updated the PR description since it does not fix the issue you filed. I'll ask you for clarifications in the issue page.

**Comment by tariq1890** [comment]
@gouthamve Can you have a look at this? I have addressed your comments

**Comment by tariq1890** [comment]
@gouthamve Ping!

**Comment by gouthamve** [comment]
👍 Thanks!

**Pull request merged**